storage/tools/dump_tree/main.go

```diff
@@ -0,0 +1,201 @@
+		glog.Fatalf("SequenceBatch got: %d sequenced, want: %d", got, want)
+	}
+}
+
+func getPrivateKey(pemPath, pemPassword string) (*any.Any, crypto.Signer) {
+	pemSigner, err := keys.NewFromPrivatePEM(pemPath, pemPassword)
+	if err != nil {
+		glog.Fatalf("NewFromPrivatePEM(): %v", err)
+	}
+	pemDer, err := keys.MarshalPrivateKey(pemSigner)
+	if err != nil {
+		glog.Fatalf("MarshalPrivateKey(): %v", err)
+	}
+	anyPrivKey, err := ptypes.MarshalAny(&keyspb.PrivateKey{Der: pemDer})
+	if err != nil {
+		glog.Fatalf("MarshalAny(%v): %v", pemDer, err)
+	}
+
+	return anyPrivKey, pemSigner
+}
+
+func getPublicKey(pem string) []byte {
+	key, err := keys.NewFromPublicPEM(pem)
+	if err != nil {
+		panic(err)
+	}
+
+	der, err := x509.MarshalPKIXPublicKey(key)
+	if err != nil {
+		panic(err)
+	}
+	return der
+}
+
+
+func createTree(as storage.AdminStorage) (*trillian.Tree, crypto.Signer) {
+	atx, err := as.Begin(context.TODO())
+	if err != nil {
+		glog.Fatalf("Begin admin TX: %v", err)
+	}
+
+	privKey, cSigner := getPrivateKey(logPrivKeyPEM, "towel")
+	pubKey := getPublicKey(logPubKeyPEM)
+
+	tree := trillian.Tree{
+		TreeType:           trillian.TreeType_LOG,
+		TreeState:          trillian.TreeState_ACTIVE,
+		HashAlgorithm:      sigpb.DigitallySigned_SHA256,
+		HashStrategy:       trillian.HashStrategy_RFC6962_SHA256,
+		SignatureAlgorithm: sigpb.DigitallySigned_ECDSA,
+		PrivateKey:         privKey,
+		PublicKey:          &keyspb.PublicKey{Der: pubKey},
+		MaxRootDuration:    ptypes.DurationProto(0 * time.Millisecond),
+	}
+	t, err := atx.CreateTree(context.TODO(), &tree)
+	if err != nil {
+		glog.Fatalf("Create tree: %v", err)
+	}
+	if err := atx.Commit(); err != nil {
+		glog.Fatalf("Commit admin TX: %v", err)
+	}
+
+	return t, cSigner
+}
+
+func main() {
+	flag.Parse()
+
+	glog.Info("Initializing memory log storage")
+	ls := memory.NewLogStorage(monitoring.InertMetricFactory{})
+	as := memory.NewAdminStorage(ls)
+	tree, cSigner := createTree(as)
+
+	seq := log.NewSequencer(rfc6962.DefaultHasher,
+		util.SystemTimeSource{},
+		ls,
+		&tc.Signer{Signer: cSigner, Hash: crypto.SHA256},
+		nil,
+		quota.Noop())
+
+	sequence(tree.TreeId, seq, 0)
+
+	glog.Info("Queuing work")
+	for l := 0; l < *treeSizeFlag; l++ {
+		glog.V(1).Infof("Queuing leaf %d", l)
+
+		leafData := []byte(fmt.Sprintf(*leafDataFormatFlag, l))
+		tx, err := ls.BeginForTree(context.TODO(), tree.TreeId)
+		if err != nil {
+			glog.Fatalf("BeginForTree got: %v, want: nil", err)
+		}
+
+		hash := sha256.Sum256(leafData)
+		lh := []byte(hash[:])
+		leaf := trillian.LogLeaf{LeafValue: leafData, LeafIdentityHash: lh, MerkleLeafHash: lh}
+		leaves := []*trillian.LogLeaf{&leaf}
+
+		if _, err := tx.QueueLeaves(context.TODO(), leaves, time.Now()); err != nil {
+			glog.Fatalf("QueueLeaves got: %v, want: nil", err)
+		}
+
+		if err := tx.Commit(); err != nil {
+			glog.Fatalf("Queueleaves TX commit: %v", err)
+		}
+
+		if l > 0 && l%*batchSizeFlag == 0 {
```

Comment:
Fixed by moving the leftover call to sequencing into this method so it's complete on return.